### ISSUE TITLE
Clear date selection for optional fields

### ISSUE THREAD

**Issue by junzhengca** [aboveyunhai/chakra-dayzed-datepicker#9]
How do I clear the date selection or start with an empty input? This is quite useful for optional fields. For example I might want user to specify an empty date to indicate there is no expiry date.

Seems like onDateChange have to set a Date object. Prehaps we can make it `(date: Date | null) => void`?

![image](https://user-images.githubusercontent.com/7855724/170501569-d28b68d3-7490-488a-a2c4-e4d2b37dc5a8.png)

**Comment by aboveyunhai** [comment]
I didn't understand your question properly, can u rephrase the question a little bit? 
Do you mean something like Range Picker since you mention no expiry date

For empty date, I think you can just do `setDate(undefined)`, assume you have the hook `const [date, setDate] = useState()`
or you want something like a `clear button` to be inside the date picker panel.

**Comment by junzhengca** [comment]
This indeed works, thanks a lot for the response. @aboveyunhai maybe consider adding a clear button, but for my usecase I can just create one myself externally.

For reference if anyone else encounters the same problem

```ts
const [date, setDate] = useState<Date>();

// then do this to reset to empty
setDate(undefined);
```

I will close this issue.

**Comment by barrymichaeldoyle** [comment]
Agreed, it would be really nice to have a `clearable` prop which will render a clear button that will set the date to undefined.

**Comment by natecalc** [comment]
I too would love to see a clear button inside as a prop. Thanks! :)
